### ISSUE TITLE
[Question] Subtype language selection

### ISSUE THREAD

**Issue by bushrang3r** [florisboard/florisboard#140]
Great to see a new opensource keyboard on the scene 👍

I have installed Japanese as a language on my phone but am unable to add it as a subtype in FlorisBoard settings. Ir simply doesn't show as an option.
I wonder if I'm missing something, or if it's not possible at the moment.
I need this to be able to display the language selector button on the keyboard.
Thanks

**Comment by patrickgold** [comment]
No, besides Arabic and Persian FlorisBoard does currently not support any non-latin languages. FlorisBoard's set of languages will definitely expand, but I think that adding Japanese won't happen until after v0.4.0. I will add it to the corresponding milestone.

**Comment by bushrang3r** [comment]
Thanks Patrick,
Understood. No worries. And thanks.
I actually already have a Japanese keyboard installed but switching between keyboards on Android is just not as quick as  switching languages 'on the fly' with the language switching button. But I can live with it for now.

**Comment by lethallunatic** [comment]
@bushrang3r 
I have reported an issue #168 to @patrickgold regarding the spacebar for switching languages. You can however use the spacebar to change to another keyboard, depending on the other keyboard you may also switch back to florisboard.

**Comment by MTSYuki** [comment]
A japanese input would be wonderful.
Currently I use Mozc for Android. It's UI the same as Gboard, also open source but is not updated since 4/10/19.

Is gonna be a hard work to put suggestions, japanese language use a high number of characters.

**Comment by Hayleia** [comment]
I just tried Mozc to get an idea of how this is supposed to work (I'm not a Japanese language user at all). From what I see, the layout is Latin, Latin letters are converted to hiragana only, and if the user intended to write katakana or kanji, then there is a suggestion bar at the top to convert hiragana to katakana/kanji?

The part where latin characters are converted to hiragana is doable pretty easily now (see video (yes i'm definitely typing in canadian french)), but the part where hiragana can be converted to katakana/kanji from the smart bar would probably require autocorrect/spellcheck to be stable and then it would only be a matter of using the correct dictionary I guess?
https://user-images.githubusercontent.com/7904776/119452735-f9194a80-bd36-11eb-88e8-8c5962478b46.mp4

**Comment by MTSYuki** [comment]
@Hayleia That exactly how the Mozc works, yes, but there another layout for typing.

One of this is the 12 Keys, which is very similar to older phones (The only diference is the way you type the characters, you need to slide, instead of pressing the same button)

Another suggestion is, instead to use suggestions bar, place a button to change Hiragana to Katakana/Kanji like on PCs (The key to change the characters is Space)

**Comment by Waelwindows** [comment]
While it may be easier to implement it like this for now, I think it's better to focus on the 12 key flick style as it's the input method that most native speakers use.

**Comment by MTSYuki** [comment]
Although I'm not a native japanese speaker, I use the standard Latin layout.

Writing Hiragana and using suggestions to type Kanji or Katakana seems pretty good to me, as find the Kanji can be a bit tedius pressing the same button (The con of the PC layout).

The 12 keys, in my experience, can be confusing at first, especially for those that still learning, but when continually using, practicing and muscle memory, I think that can be a great usage for the user.

I recommend implement the 12 Keys and latin layout with the option to change the way to write Katakana/Kanji to some button or using the suggestions.

@Waelwindows I agree with you, if you see japanese people typing in their phones, most of the users use the 12 keys.

**Comment by Hayleia** [comment]
I'm linking #902 because it is also asking for flick layout support (but not necessarily applied to japanese) so it can be taken into account when thinking about flick layouts.

**Comment by RickyM7** [comment]
I believe that *[this project](https://github.com/google/mozc)* can help a lot in the implementation of this layout. It has the 12-key layout and others as well, I recommend you take a look.
